### ISSUE TITLE
tf.keras.layers.DepthwiseConv2D and tf.keras.layers.SeparableConv2D does not handle kernel regularizer correctly.

### ISSUE THREAD

**Issue by jyhpsycho** [tensorflow/tensorflow#41093]
Please go to Stack Overflow for help and support:

https://stackoverflow.com/questions/tagged/tensorflow

If you open a GitHub issue, here is our policy:

1.  It must be a bug, a feature request, or a significant problem with the
    documentation (for small docs fixes please send a PR instead).
2.  The form below must be filled out.
3.  It shouldn't be a TensorBoard issue. Those go
    [here](https://github.com/tensorflow/tensorboard/issues).

**Here's why we have that policy**: TensorFlow developers respond to issues. We want to focus on work that benefits the whole community, e.g., fixing bugs and adding features. Support only helps individuals. GitHub also notifies thousands of people when issues are filed. We want them to see you communicating an interesting problem, rather than being redirected to Stack Overflow.

------------------------

### System information

-   **Have I written custom code (as opposed to using a stock example script
    provided in TensorFlow)**: No
-   **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: CentOS 7.3
-   **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue
    happens on a mobile device**: NA
-   **TensorFlow installed from (source or binary)**: binary, from anaconda
-   **TensorFlow version (use command below)**: 2.2.0
-   **Python version**: 3.7.7
-   **Bazel version (if compiling from source)**: NA
-   **GCC/Compiler version (if compiling from source)**: NA
-   **CUDA/cuDNN version**: 10.1 / 7.6.5
-   **GPU model and memory**: NVIDIA Tesla V100 / 32GB
-   **Exact command to reproduce**: Source code below

You can collect some of this information using our environment capture script:

https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh

You can obtain the TensorFlow version with:

```bash
python -c "import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"
```

### Describe the problem
Describe the problem clearly here. Be sure to convey here why it's a bug in TensorFlow or a feature request.

kernel_regularizer parameter on tf.keras.layers.DepthwiseConv2D and  tf.keras.layers.SeparableConv2D does not register the loss to model losses, despite of bias_regularizer works. I'll describe full test code below.

### Source code / logs
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached. Try to provide a reproducible test case that is the bare minimum necessary to generate the problem.

Expected case 1: Dense layer
```python
m = tf.keras.models.Sequential()
m.add(tf.keras.layers.Dense(3, input_dim=5, kernel_regularizer=tf.keras.regularizers.l2(1.0E-01), bias_regularizer=tf.keras.regularizers.l2(1.0E-01)))
m.add(tf.keras.layers.Dense(3, input_dim=5, kernel_regularizer=tf.keras.regularizers.l2(1.0E-01)))
print(m.losses)
```
Output:
```
[<tf.Tensor: shape=(), dtype=float32, numpy=0.19434488>,
 <tf.Tensor: shape=(), dtype=float32, numpy=0.0>,
 <tf.Tensor: shape=(), dtype=float32, numpy=0.22874394>]
```

Expected case 2: Conv2D layer
```python
m = tf.keras.models.Sequential()
m.add(tf.keras.layers.Input(32, 32, 3))
m.add(tf.keras.layers.Conv2D(3, 3, kernel_regularizer=tf.keras.regularizers.l2(1.0E-01), bias_regularizer=tf.keras.regularizers.l2(1.0E-01)))
m.add(tf.keras.layers.Conv2D(3, 3, kernel_regularizer=tf.keras.regularizers.l2(1.0E-01)))
print(m.losses)
```
Output:
```
[<tf.Tensor: shape=(), dtype=float32, numpy=0.29766324>,
 <tf.Tensor: shape=(), dtype=float32, numpy=0.0>,
 <tf.Tensor: shape=(), dtype=float32, numpy=0.7301255>]
```

Expected case 3: ConvLSTM2D layer
```python
m = tf.keras.models.Sequential()
m.add(tf.keras.layers.Input(3, 32, 32, 3))
m.add(tf.keras.layers.ConvLSTM2D(3, 3, return_sequences=True, kernel_regularizer=tf.keras.regularizers.l2(1.0E-01), bias_regularizer=tf.keras.regularizers.l2(1.0E-01)))
m.add(tf.keras.layers.ConvLSTM2D(3, 3, kernel_regularizer=tf.keras.regularizers.l2(1.0E-01)))
print(m.losses)
```
Output:
```
[<tf.Tensor: shape=(), dtype=float32, numpy=0.47271797>,
 <tf.Tensor: shape=(), dtype=float32, numpy=0.3>,
 <tf.Tensor: shape=(), dtype=float32, numpy=0.49562755>]
```

Does not expected case 1: DepthwiseConv2D layer
```python
m = tf.keras.models.Sequential()
m.add(tf.keras.layers.Input(32, 32, 3))
m.add(tf.keras.layers.DepthwiseConv2D(3, 3, kernel_regularizer=tf.keras.regularizers.l2(1.0E-01), bias_regularizer=tf.keras.regularizers.l2(1.0E-01)))
m.add(tf.keras.layers.DepthwiseConv2D(3, 3, kernel_regularizer=tf.keras.regularizers.l2(1.0E-01)))
print(m.losses)
```
Output:
```
[<tf.Tensor: shape=(), dtype=float32, numpy=0.0>]
```

Does not expected case 2: SeparableConv2D layer
```python
m = tf.keras.models.Sequential()
m.add(tf.keras.layers.Input(32, 32, 3))
m.add(tf.keras.layers.SeparableConv2D(3, 3, kernel_regularizer=tf.keras.regularizers.l2(1.0E-01), bias_regularizer=tf.keras.regularizers.l2(1.0E-01)))
m.add(tf.keras.layers.SeparableConv2D(3, 3, kernel_regularizer=tf.keras.regularizers.l2(1.0E-01)))
print(m.losses)
```
Output:
```
[<tf.Tensor: shape=(), dtype=float32, numpy=0.0>]
```

Am I have a mistake using regularizer with these layers?

**Comment by jyhpsycho** [comment]
SeparableConv1D also has same issue.

Case Conv1D:
```python
m = tf.keras.models.Sequential()
m.add(tf.keras.layers.Input(32, 3))
m.add(tf.keras.layers.Conv1D(3, 3, kernel_regularizer=tf.keras.regularizers.l2(1.0E-01), bias_regularizer=tf.keras.regularizers.l2(1.0E-01)))
m.add(tf.keras.layers.Conv1D(3, 3, kernel_regularizer=tf.keras.regularizers.l2(1.0E-01)))
print(m.losses)
```

Output:
```
[<tf.Tensor: shape=(), dtype=float32, numpy=0.30436406>,
 <tf.Tensor: shape=(), dtype=float32, numpy=0.0>,
 <tf.Tensor: shape=(), dtype=float32, numpy=0.31131223>]
```

Case SeparableConv1D:
```python
m = tf.keras.models.Sequential()
m.add(tf.keras.layers.Input(32, 3))
m.add(tf.keras.layers.SeparableConv1D(3, 3, kernel_regularizer=tf.keras.regularizers.l2(1.0E-01), bias_regularizer=tf.keras.regularizers.l2(1.0E-01)))
m.add(tf.keras.layers.SeparableConv1D(3, 3, kernel_regularizer=tf.keras.regularizers.l2(1.0E-01)))
print(m.losses)
```

Output:
```
[<tf.Tensor: shape=(), dtype=float32, numpy=0.0>]
```

**Comment by edwardyehuang** [comment]
@jyhpsycho  Have you test it on tf.keras.layers.Conv2D and tf.nn.depthwise_conv2d ?

**Comment by ravikyram** [comment]
@jyhpsycho 

Request you to share colab link or complete code snippet with supporting files to reproduce the issue in our environment.It helps us in localizing the issue faster.Thanks!

**Comment by jyhpsycho** [comment]
@ravikyram 

Oh, I found that some type on my examples(shape of input layer should be tuple, but not). I'll upload IPython notebook file and converted .py script for reproduce it.

[41093.zip](https://github.com/tensorflow/tensorflow/files/4882572/41093.zip)

**Comment by ravikyram** [comment]
I have tried in colab with TF versions 2.2, nightly versions and was able to reproduce the issue.Please, find the gist [here](https://colab.research.google.com/gist/ravikyram/13d016c1c285c1c3a92bfadc9efaa5fb/untitled92.ipynb).Thanks!

**Comment by mkarpushin-upstride** [comment]
Please pay attention that according to the docs there is no `kernel_regularizer` parameter in `DepthwiseConv2D`, but there is `depthwise_regularizer`: https://www.tensorflow.org/api_docs/python/tf/keras/layers/DepthwiseConv2D
The problem would be then that while keras accepts the both, it does not apply `kernel_regularizer` without actually warning the user.

**Comment by jyhpsycho** [comment]
@mkarpushin-upstride The document seems to be changed after v2.3 release. On v2.2 there's no `depthwise_regularizer`. The end-user cannot know what the option name should be used for specifying regularizer at that time, even if `depthwise_regularizer` works with v2.2.
https://www.tensorflow.org/versions/r2.2/api_docs/python/tf/keras/layers/ConvLSTM2D

BTW, if `depthwise_regularizer` works on 2.2 and further version, this issue can be closed.

**Comment by jyhpsycho** [comment]
https://colab.research.google.com/gist/jyhpsycho/3fd5a2d8ff6071c013137a69af6acb18/untitled92.ipynb

I checked the `depthwise_regularizer` option (based on @ravikyram's colab code) and it seems to work with TF 2.2. @mkarpushin-upstride Thank you for clarification.

If there's any problem with test code, feel free to reopen this issue.
